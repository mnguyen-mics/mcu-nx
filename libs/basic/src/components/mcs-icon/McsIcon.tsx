import { HomeOutlined } from '@ant-design/icons';
import React from 'react';

export type McsIconType =
  | 'adGroups'
  | 'ads'
  | 'automation'
  | 'bell'
  | 'bolt'
  | 'check-rounded-inverted'
  | 'check-rounded'
  | 'check'
  | 'chevron-right'
  | 'chevron'
  | 'close-big'
  | 'close-rounded'
  | 'close'
  | 'code'
  | 'creative'
  | 'data'
  | 'delete'
  | 'display'
  | 'dots'
  | 'download'
  | 'email-inverted'
  | 'email'
  | 'extend'
  | 'filters'
  | 'full-users'
  | 'file'
  | 'gears'
  | 'goals-rounded'
  | 'goals'
  | 'home'
  | 'image'
  | 'info'
  | 'laptop'
  | 'library'
  | 'magnifier'
  | 'menu-close'
  | 'minus'
  | 'optimization'
  | 'options'
  | 'partitions'
  | 'pause'
  | 'pen'
  | 'phone'
  | 'play'
  | 'plus'
  | 'query'
  | 'question'
  | 'refresh'
  | 'settings'
  | 'smartphone'
  | 'status'
  | 'tablet'
  | 'user'
  | 'users'
  | 'user-query'
  | 'user-pixel'
  | 'user-list'
  | 'video'
  | 'warning';

const antdIcons: McsIconType[] = ['home'];

export interface McsIconProps {
  type: McsIconType;
  styleIcon?: React.CSSProperties;
}

class McsIcon extends React.Component<
  McsIconProps & React.DetailedHTMLProps<React.HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>
> {
  render() {
    const { type, styleIcon, className, ...rest } = this.props;

    if (antdIcons.includes(type)) {
      if (type === 'home') return <HomeOutlined className={'mcs-icon'} style={rest.style} />;
      else return <span />;
    } else
      return (
        <span className={`mcs-icon ${className ? className : ''}`} {...rest}>
          <i className={`mcs-${type}`} style={styleIcon ? styleIcon : {}} />
        </span>
      );
  }
}

export default McsIcon;
