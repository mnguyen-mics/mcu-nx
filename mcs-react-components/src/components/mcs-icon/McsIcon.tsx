import * as React from 'react';

export type McsIconType =
  'adGroups' |
  'ads' |
  'automation' |
  'bell' |
  'bolt' |
  'check-rounded-inverted' |
  'check-rounded' |
  'check' |
  'chevron-right' |
  'chevron' |
  'close-big' |
  'close-rounded' |
  'close' |
  'code' |
  'creative' |
  'data' |
  'delete' |
  'display' |
  'dots' |
  'download' |
  'email-inverted' |
  'email' |
  'extend' |
  'filters' |
  'full-users' |
  'file' |
  'gears' |
  'goals-rounded' |
  'goals' |
  'image' |
  'info' |
  'laptop' |
  'library' |
  'magnifier' |
  'menu-close' |
  'minus' |
  'optimization' |
  'options' |
  'partitions' |
  'pause' |
  'pen' |
  'phone' |
  'play' |
  'plus' |
  'query' |
  'question' |
  'refresh' |
  'settings' |
  'smartphone' |
  'status' |
  'tablet' |
  'user' |
  'users' |
  'user-query' |
  'user-pixel' |
  'user-list' |
  'video' |
  'warning';

export interface McsIconProps {
  type: McsIconType;
  styleIcon?: React.CSSProperties;
}

class McsIcon extends React.Component<
  McsIconProps &
    React.DetailedHTMLProps<
      React.HTMLAttributes<HTMLSpanElement>,
      HTMLSpanElement
    >
> {
  render() {
    const { type, styleIcon, className, ...rest } = this.props;
    return (
      <span className={`mcs-icon ${className ? className : ''}`} {...rest}>
        <i className={`mcs-${type}`} style={styleIcon ? styleIcon : {}}/>
      </span>
    );
  }
}

export default McsIcon;