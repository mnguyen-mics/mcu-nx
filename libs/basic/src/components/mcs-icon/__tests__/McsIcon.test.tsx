import 'jest';
import React from 'react';
import TestRenderer from 'react-test-renderer';

import McsIcon from '../McsIcon';

it('renders an user icon', () => {
  const component = TestRenderer.create(<McsIcon type='user' />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
