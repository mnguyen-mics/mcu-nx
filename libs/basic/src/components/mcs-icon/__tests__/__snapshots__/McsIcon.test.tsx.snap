// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an user icon 1`] = `
<span
  className="mcs-icon "
>
  <i
    className="mcs-user"
    style={Object {}}
  />
</span>
`;
